backend/chromeApiServer/src/main/java/com/canvas/service/EvaluationService.java

```diff
@@ -0,0 +1,157 @@
+
+import com.canvas.service.helperServices.CanvasClientService;
+import com.canvas.service.helperServices.FileService;
+import com.canvas.service.helperServices.ProcessExecutor;
+import com.canvas.service.models.CommandOutput;
+import com.canvas.service.models.ExtensionUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles the Student side of the evaluation process.
+ */
+@Service
+public class EvaluationService {
+    private final FileService fileService;
+    private final CanvasClientService canvasClientService;
+    String MAKEFILE = "makefile";
+
+    /**
+     * Constructor for creating our instances of FileService and CanvasClientService
+     *
+     * @param fileService         File service instance
+     * @param canvasClientService Canvas client service instance
+     */
+    @Autowired
+    public EvaluationService(FileService fileService, CanvasClientService canvasClientService) {
+        this.fileService = fileService;
+        this.canvasClientService = canvasClientService;
+    }
+
+    /**
+     * This method will compile the student code. The files are extracted
+     * and the program is compiled.
+     *
+     * @param user  User object holding all components associated with a user
+     * @param files The files to compile
+     * @return      return a CommandOutput response
+     */
+    public ResponseEntity<CommandOutput> compileStudentCodeFile(ExtensionUser user, MultipartFile[] files) {
+
+
+        // Retrieve file json from Canvas
+        byte[] makefileBytes = new byte[0];
+        try {
+            makefileBytes = canvasClientService.fetchFileUnderCourseAssignmentFolder(user, MAKEFILE);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        // Write makefile to file
+        fileService.writeFileFromBytes(MAKEFILE, makefileBytes, user.getUserId());
+
+        // Write submitted code files
+        for (MultipartFile file : files) {
+            fileService.writeFileFromMultipart(file, user.getUserId());
+        }
+
+        // Compile the files and grab output
+        ProcessExecutor processExecutor = new ProcessExecutor(new String[]{"make"}, fileService.getFileDirectory(user.getUserId()));
+        boolean compileSuccess = processExecutor.executeProcess();
+        String output;
+        if (compileSuccess) {
+            output = "Your program compiled successfully!";
+        } else {
+            output = processExecutor.getProcessOutput();
+        }
+
+        // Cleanup
+        fileService.deleteDirectory(user.getUserId());
+
+        // Generate response
+        CommandOutput commandOutput = new CommandOutput(compileSuccess, output);
```

Comment:
[If we call compileCodeFiles instead](https://github.com/Canvas-Code-Capstone/Canvas-Code/pull/18/files#r1029950984) we could save the CommandOutput obj that gets returned, so we could remove this line too